ext/session/mod_files.c

```diff
@@ -165,7 +165,7 @@ static void ps_files_open(ps_files *data, /* const */ zend_string *key)
 
 		ps_files_close(data);
 
-		if (php_session_valid_key(ZSTR_VAL(key)) == FAILURE) {
+		if (php_session_valid_key(key) == FAILURE) {
 			php_error_docref(NULL, E_WARNING, "Session ID is too long or contains illegal characters. Only the A-Z, a-z, 0-9, \"-\", and \",\" characters are allowed");
 			return;
 		}
```

Comment:
It should not be possible to have an empty `key` here.

Please add an assertion.

Reply:
I kept the previous logic when it's treated as an error.